### PR TITLE
[Infra] Use macOS 12 and Xcode 14 in GHA

### PR DESCRIPTION
**Pull request by ncooke3**
Successor to https://github.com/firebase/quickstart-ios/pull/1424

### CODE REVIEW

**Comment by ncooke3** [comment]
@paulb777, we don't use a matrix with Xcode versions for all jobs... should we?

**Comment by paulb777** [comment]
Yes, but not a high priority

**Comment by ncooke3** [comment]
Sorry @morganchen12, didn't realize you had this underway at #1425. Feel free to take this over. I was waiting til after Firebase 10 stuff to resume it

**Review by paulb777 — APPROVED** [review]

File: abtesting/Shared/UITests.swift [hunk 1]
```diff
@@ -68,7 +68,13 @@ class UITests: XCTestCase {
       let texts = ["Getting Started with Firebase", "An Introduction to Firebase"]
     #endif
     for text in texts {
-      XCTAssertTrue(app.staticTexts[text].isHittable, "Text '\(text)' is missing from view.")
+      let element = app.staticTexts[text]
+      // TODO: This if should be removed.
```
Comment (ncooke3) [inline abtesting/Shared/UITests.swift:72]: Flagging this in case you meant to remove before merging.

Comment (morganchen12) [reply abtesting/Shared/UITests.swift:72]: This is from https://github.com/firebase/quickstart-ios/pull/1456. Tests fail without it on GHA but not locally. I'll try if it's fixed on Xcode 14 though.

Comment (morganchen12) [reply abtesting/Shared/UITests.swift:72]: Still present on Xcode 14 :(

**Pull request merged**